controller/sharding/sharding.go

```diff
@@ -73,12 +74,14 @@ func GetClusterFilter(db db.ArgoDB, distributionFunction DistributionFunction, s
 
 // GetDistributionFunction returns which DistributionFunction should be used based on the passed algorithm and
 // the current datas.
-func GetDistributionFunction(db db.ArgoDB, shardingAlgorithm string) DistributionFunction {
-	log.Infof("Using filter function:  %s", shardingAlgorithm)
+func GetDistributionFunction(db db.ArgoDB, clusters clusterAccessor, shardingAlgorithm string) DistributionFunction {
```

Comment:
ok, got it, I have pushed last changes to fix that.
I need to have a look at the case where the number replicas is changing.

Reply:
For StatefulSets, the pods do get restarted whenever we want to update the number of replicas. But for deployments, we read the replicas directly from the deployments without any pod restart.

We should incorporate changes for change in replicas without pod restart too. If that is going to be huge change, it might be worth moving that change to a new PR to make it easier for reviewing.